### DISCUSSION TITLE
Can't add subdevice for an general IR Remote

### DISCUSSION THREAD

**Discussion by alexantao** [make-all/tuya-local#2208] [Q&A]
Hi, could somebody point me the way, if it's already answered ? I can't find. 

I have those general IR Remote Control bought on Ali.
I can add it to Local Tuya, through the integration UI. It shows on device list on Local Tuya and it has just one entity, the ON/OFF switch. 

I want to add the devices already attached to it (air conditioner, TV, etc), but I have 2 situations:
  1) If the IR is added, when I select the subdevicem it asks fot the HUB it's connected, but the device is not shown on the list. Even trying to mannually add it does not work.
  2) If the IR is not added to HA, when adding the subdevice, it is shown on the HUB list, but it does not work either. I also tried mannually, changind all the id's combination I could. 

Is there a way I can add then ????

Thanks,

**Comment by make-all** [comment]
You cannot add the subdevices for remote controls because they do not return any data to use in entities.
You need to add the IR device and use the remote entity to learn and control the other devices from tuya-local.

It sounds like your device is an unsupported device though and currently is only adding as a simple switch.

**Reply by alexantao** [reply]
I was doing a lot of search here. Unassembled my IR and found out it has a TYWE3S module.
I don´t  know how compatible it is with HA, but seems I can flash Tasmota-IR on it. So, if I cannot control natively, I will try to flash it. 
Thanks.
